habitat-lab/habitat/tasks/rearrange/rearrange_task.py

```diff
@@ -211,12 +219,24 @@ def _filter_agent_position(start_pos, start_rot):
                     return np.all(distances > self._min_distance_start_agents)
 
                 filter_agent_position = _filter_agent_position
-            (
-                articulated_agent_pos,
-                articulated_agent_rot,
-            ) = self._sim.set_articulated_agent_base_to_random_point(
-                agent_idx=agent_idx, filter_func=filter_agent_position
-            )
+            # (
+            #     articulated_agent_pos,
+            #     articulated_agent_rot,
+            # ) = self._sim.set_articulated_agent_base_to_random_point(
+            #     agent_idx=agent_idx, filter_func=filter_agent_position
+            # )
+            episode_id = self._sim.ep_info.episode_id
+            agents = self._robot_config[episode_id]['agents']
+            for agent in agents:
+                if agent['agent_idx'] == agent_idx:
+                    (
+                        articulated_agent_pos,
+                        articulated_agent_rot,
+                    ) = (
+                        np.array(agent['start_pos']),
+                        agent['start_rot']
+                    )
+                    break
```

Comment:
Since rearrage_task.py is a main functional class to run all rearrange-related tasks (social rearrange, legacy rearrange, etc.) We need a flag to choose between using randomized init. states or saved states.